maxbin2 2.2.5 (new formula)

suujia opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

suujia commented:
Still testing! 
From read me: 
==== 
```
Enter src directory under MaxBin and "make" it.

=== Prerequisite ===
1. g++ (c++11 or higher), make
2. Perl5 with CPAN, LWP::Simple, and FindBin modules
   (./autobuild_auxiliary will use CPAN to check and install the other two Perl modules)
3. (for automatic downloading and installing auxiliary software) curl, unzip, which

   - The manual way. Auxiliary packages include Bowtie2, FragGeneScan, IDBA-UD, and Hmmer3. Please follow the instructions in the software packages to compile and install them.

	* Bowtie2 (tested version: bowtie2-2.2.3)
	  http://bowtie-bio.sourceforge.net/bowtie2/index.shtml
	* FragGeneScan (tested version: FragGeneScan 1.30)
	  http://omics.informatics.indiana.edu/FragGeneScan/
	* Hmmer3 (tested version: HMMER 3.1b1 64 bit)
	  http://hmmer.janelia.org/
	* IDBA-UD (tested version: IDBA-UD 1.1.1)
	  http://i.cs.hku.hk/~alse/hkubrg/projects/idba_ud/
```

sjackman commented on Formula/maxbin2.rb:
```diff
@@ -0,0 +1,39 @@
+    # cite Wu_2016: "https://doi.org/10.1093/bioinformatics/btv638"
+    desc "an automated binning algorithm to recover genomes from multiple metagenomic datasets"
+    homepage "https://bioconda.github.io/recipes/maxbin2/README.html"
+    url "https://sourceforge.net/projects/maxbin2/files/MaxBin-2.2.5.tar.gz/download"
```
Try…
```ruby
url "https://downloads.sourceforge.net/project/maxbin2/MaxBin-2.2.5.tar.gz"
```

sjackman commented on Formula/maxbin2.rb:
```diff
@@ -0,0 +1,39 @@
+    resource "fraggenescan" do 
+        url "https://sourceforge.net/projects/fraggenescan/files/FragGeneScan1.31.tar.gz/download"
+        sha256 "cd3212d0f148218eb3b17d24fcd1fc897fb9fee9b2c902682edde29f895f426c"
+    end
```
Could `fraggenescan` be a separate formula rather than a resource?

sjackman commented on Formula/maxbin2.rb:
```diff
@@ -0,0 +1,39 @@
+    end
+
+    def install
+      ENV.prepend "PERL5LIB", Formula["bioperl"].libexec/"lib/perl5"
```
Use `ENV.prepend.path` to ensure that the separator is `:` rather than a space

sjackman commented on Formula/maxbin2.rb:
```diff
@@ -0,0 +1,39 @@
+
+      cd "src" do
+          system "make"
+      end
```
After running `make` you'll need to install the executables that it built. If there's no `make install` target, then use…
```ruby
bin.install "maxbin2", "any-other-executables-go-here"
```

sjackman commented:
Check the output of `brew style maxbin2` and `brew audit --strict --new-formula maxbin2`

tseemann commented:
```
==> brew readall --aliases brewsci/bio
Error: No available formula with the name "fraggenescan" 
Exited with code 1
```

suujia commented:
Hmm. not really sure what to do about this error. 
```
==> brew install --verbose --build-bottle brewsci/bio/maxbin2
==> Installing maxbin2 from brewsci/bio
==> Downloading https://downloads.sourceforge.net/project/maxbin2/MaxBin-2.2.5.tar.gz
Already downloaded: /home/linuxbrew/.cache/Homebrew/maxbin2--2.2.5.tar.gz
==> Verifying maxbin2--2.2.5.tar.gz checksum
gunzip -N -- /tmp/d20180830-6755-nyyvzs/MaxBin-2.2.5.tar.gz
tar xf /tmp/d20180830-6755-nyyvzs/MaxBin-2.2.5.tar -C /tmp/d20180830-6755-11lpicc
cp -pR /tmp/d20180830-6755-11lpicc/MaxBin-2.2.5/. /tmp/maxbin2-20180830-6755-1lxk5r6/MaxBin-2.2.5
Error: wrong number of arguments (given 0, expected 2..3)
```

sjackman commented on Formula/maxbin2.rb:
```diff
@@ -0,0 +1,30 @@
+  depends_on "perl" unless OS.mac?
+
+  def install
+    ENV.prepend.path "PERL5LIB", Formula["bioperl"].libexec/"lib/perl5"
```
There's a typo here. `prepend.path` should be `prepend_path`

sjackman commented on Formula/maxbin2.rb:
```diff
@@ -0,0 +1,30 @@
+    cd "src" do
+      bin.install Dir["*"]
+    end
+    bin.install_symlink Dir["#{prefix}/run_MaxBin.pl"]
```
`Dir` is only needed when using a wildcard like `*`
```ruby
    bin.install_symlink prefix/"run_MaxBin.pl"
```

suujia commented:
Not sure how to fix this! 
```
Program not built. Please run "make" under src directory to build MaxBin program.
```
I've tried it with  bin.install Dir["*"] following the system"make" as well. This is the makefile in src: 
```
CC:=g++
CFLAGS:=-c -Wall -g -std=c++11
LDFLAGS:=-pthread -std=c++11
INC:=-I ./
SOURCES:=AbundanceLoader.cpp EManager.cpp EucDist.cpp fastaReader.cpp \
	KendallDist.cpp kmerMap.cpp logger.cpp main.cpp ManhattanDist.cpp \
	MinManhattanDist.cpp Profiler.cpp quickSort.cpp SpearmanDist.cpp \
	AbstractDist.cpp NormalDistribution.cpp ThreadPool.cpp
OBJECTS:=$(SOURCES:.cpp=.o)
EXECUTABLE:=MaxBin
all: $(EXECUTABLE)
$(EXECUTABLE): $(OBJECTS)
	$(CC) $(LDFLAGS) -o $(EXECUTABLE) $(OBJECTS) $(INC) $(LDFLAGS)
$(OBJECTS): %.o: %.cpp
	$(CC) $(CFLAGS) $< -o $@ $(INC)
clean:
	rm $(EXECUTABLE) $(OBJECTS)
```
and the readme just says
```
Enter src directory under MaxBin and "make" it.
```


### Other changes I've made: 
added LWP::Simple based on this issue: https://github.com/brewsci/homebrew-bio/pull/267#issuecomment-388999848
added IO::Socket::SSL based on this issue: https://github.com/brewsci/homebrew-bio/pull/267#issuecomment-397708020

suujia commented:
Build passed on linux and travis, fails on mac for following error: 
```
  Test::Unit::AssertionFailedError: </Usage/> expected to be =~
<"sh: line 1: 25890 Abort trap: 6           idba_ud > /dev/null 2> tmp_1536130889\nCannot run IDBA-UD. Please indicate the file directory in 'setting' file.\n">.
```

In specifications it says: 
prerequisites:  g++ (c++11 or higher)  

Should we specify anything relating to that?

sjackman commented:
```
❯❯❯ idba_ud
dyld: Library not loaded: /Users/sjackman/.homebrew/opt/gcc/lib/gcc/7/libgomp.1.dylib
  Referenced from: /Users/sjackman/.homebrew/bin/idba_ud
  Reason: image not found
```
The `idba` bottle needs to be rebuilt on macOS for `gcc`.
I've opened PR https://github.com/brewsci/homebrew-bio/pull/421

sjackman commented:
I've merged PR #421 to bump `idba`, and I've rebased this PR onto master.

sjackman commented:
Merged. Thanks again, Susie! 🏆

sjackman commented:
I've added `brew update` to `orca-5` and triggered a rebuild on Docker Hub.
